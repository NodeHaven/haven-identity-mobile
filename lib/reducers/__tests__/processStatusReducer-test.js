// Copyright (C) 2018 ConsenSys AG
//
// This file is part of uPort Mobile App.
//
// uPort Mobile App is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// uPort Mobile App is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with uPort Mobile App.  If not, see <http://www.gnu.org/licenses/>.
//
import {
  saveMessage, saveError, clearMessage, clearAllMessages, startWorking, stopWorking, failProcess, completeProcess
} from 'uPortMobile/lib/actions/processStatusActions'
import baseReducer from 'uPortMobile/lib/reducers/processStatusReducer.js'
import { toClj, toJs, hashMap } from 'mori'

const reducer = (state, action) => toJs(baseReducer(state, action))

const empty = hashMap()

const withMessages = toClj({
  unnu: {
    message: 'Woohoo',
    error: "nooooh"
  },
  sensui: {
    working: true
  }
})

it('sets message for section', () => {
  expect(reducer(empty, saveMessage('unnu', 'Looking good'))).toMatchSnapshot()
  expect(reducer(withMessages, saveMessage('unnu', 'Looking good'))).toMatchSnapshot()
  expect(reducer(withMessages, saveMessage('sensui', 'Looking good'))).toMatchSnapshot()
})

it('sets error for section', () => {
  expect(reducer(empty, saveError('unnu', 'Dang it broke'))).toMatchSnapshot()
  expect(reducer(withMessages, saveError('unnu', 'Dang it broke'))).toMatchSnapshot()
  expect(reducer(withMessages, saveError('sensui', 'Dang it broke'))).toMatchSnapshot()
})

it('clears messages for section', () => {
  expect(reducer(empty, clearMessage('unnu'))).toMatchSnapshot()
  expect(reducer(withMessages, clearMessage('unnu'))).toMatchSnapshot()
  expect(reducer(withMessages, clearMessage('sensui'))).toMatchSnapshot()
})

it('clears all messages for section', () => {
  expect(reducer(empty, clearAllMessages())).toMatchSnapshot()
  expect(reducer(withMessages, clearAllMessages())).toMatchSnapshot()
})

it('starts working for section', () => {
  expect(reducer(empty, startWorking('unnu'))).toMatchSnapshot()
  expect(reducer(withMessages, startWorking('unnu'))).toMatchSnapshot()
  expect(reducer(withMessages, startWorking('sensui'))).toMatchSnapshot()
})

it('stops working for section', () => {
  expect(reducer(empty, stopWorking('unnu'))).toMatchSnapshot()
  expect(reducer(withMessages, stopWorking('unnu'))).toMatchSnapshot()
  expect(reducer(withMessages, stopWorking('sensui'))).toMatchSnapshot()
})

it('completes process for section', () => {
  expect(reducer(empty, completeProcess('unnu'))).toMatchSnapshot()
  expect(reducer(withMessages, completeProcess('unnu'))).toMatchSnapshot()
  expect(reducer(withMessages, completeProcess('sensui'))).toMatchSnapshot()
})

it('sets error for section', () => {
  expect(reducer(empty, failProcess('unnu', 'Dang it broke'))).toMatchSnapshot()
  expect(reducer(withMessages, failProcess('unnu', 'Dang it broke'))).toMatchSnapshot()
  expect(reducer(withMessages, failProcess('sensui', 'Dang it broke'))).toMatchSnapshot()
})

it('ignores unsupported action', () => {
  expect(baseReducer(undefined, {type: 'UNSUPPORTED'})).toEqual(empty)
  expect(baseReducer(empty, {type: 'UNSUPPORTED'})).toEqual(empty)
  expect(baseReducer(withMessages, {type: 'UNSUPPORTED'})).toEqual(withMessages)
})
