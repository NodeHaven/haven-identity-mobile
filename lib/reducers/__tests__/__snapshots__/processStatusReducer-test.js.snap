// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clears all messages for section 1`] = `Object {}`;

exports[`clears all messages for section 2`] = `Object {}`;

exports[`clears messages for section 1`] = `Object {}`;

exports[`clears messages for section 2`] = `
Object {
  "sensui": Object {
    "working": true,
  },
}
`;

exports[`clears messages for section 3`] = `
Object {
  "unnu": Object {
    "error": "nooooh",
    "message": "Woohoo",
  },
}
`;

exports[`completes process for section 1`] = `
Object {
  "unnu": Object {
    "completed": true,
  },
}
`;

exports[`completes process for section 2`] = `
Object {
  "sensui": Object {
    "working": true,
  },
  "unnu": Object {
    "completed": true,
    "error": "nooooh",
  },
}
`;

exports[`completes process for section 3`] = `
Object {
  "sensui": Object {
    "completed": true,
  },
  "unnu": Object {
    "error": "nooooh",
    "message": "Woohoo",
  },
}
`;

exports[`sets error for section 1`] = `
Object {
  "unnu": Object {
    "error": "Dang it broke",
  },
}
`;

exports[`sets error for section 2`] = `
Object {
  "sensui": Object {
    "working": true,
  },
  "unnu": Object {
    "error": "Dang it broke",
    "message": "Woohoo",
  },
}
`;

exports[`sets error for section 3`] = `
Object {
  "sensui": Object {
    "error": "Dang it broke",
    "working": true,
  },
  "unnu": Object {
    "error": "nooooh",
    "message": "Woohoo",
  },
}
`;

exports[`sets error for section 4`] = `
Object {
  "unnu": Object {
    "error": "Dang it broke",
  },
}
`;

exports[`sets error for section 5`] = `
Object {
  "sensui": Object {
    "working": true,
  },
  "unnu": Object {
    "error": "Dang it broke",
  },
}
`;

exports[`sets error for section 6`] = `
Object {
  "sensui": Object {
    "error": "Dang it broke",
  },
  "unnu": Object {
    "error": "nooooh",
    "message": "Woohoo",
  },
}
`;

exports[`sets message for section 1`] = `
Object {
  "unnu": Object {
    "message": "Looking good",
  },
}
`;

exports[`sets message for section 2`] = `
Object {
  "sensui": Object {
    "working": true,
  },
  "unnu": Object {
    "message": "Looking good",
  },
}
`;

exports[`sets message for section 3`] = `
Object {
  "sensui": Object {
    "message": "Looking good",
    "working": true,
  },
  "unnu": Object {
    "error": "nooooh",
    "message": "Woohoo",
  },
}
`;

exports[`starts working for section 1`] = `
Object {
  "unnu": Object {
    "working": true,
  },
}
`;

exports[`starts working for section 2`] = `
Object {
  "sensui": Object {
    "working": true,
  },
  "unnu": Object {
    "error": "nooooh",
    "message": "Woohoo",
    "working": true,
  },
}
`;

exports[`starts working for section 3`] = `
Object {
  "sensui": Object {
    "working": true,
  },
  "unnu": Object {
    "error": "nooooh",
    "message": "Woohoo",
  },
}
`;

exports[`stops working for section 1`] = `
Object {
  "unnu": null,
}
`;

exports[`stops working for section 2`] = `
Object {
  "sensui": Object {
    "working": true,
  },
  "unnu": Object {
    "error": "nooooh",
    "message": "Woohoo",
  },
}
`;

exports[`stops working for section 3`] = `
Object {
  "sensui": Object {},
  "unnu": Object {
    "error": "nooooh",
    "message": "Woohoo",
  },
}
`;
